clio-server/src/main/scala/org/broadinstitute/clio/server/service/PersistenceService.scala

```diff
@@ -51,10 +55,59 @@ class PersistenceService private (persistenceDAO: PersistenceDAO,
       document.upsertId
     }
   }
+
+  private val countZero = Future.successful(0L)
+
+  /**
+    * Recover missing documents for a given Elasticsearch index
+    * from the source of truth.
+    *
+    * Returns the number of updates pulled & applied on success.
+    *
+    * Fails immediately if an update pulled from the source of
+    * truth fails to be applied to the search index.
+    */
+  def recoverMetadata[D <: ClioDocument: Indexable: HitReader: Decoder](
+    index: ElasticsearchIndex[D]
+  )(implicit ec: ExecutionContext): Future[Long] = {
+    searchDAO.getMostRecentDocument(index).flatMap { maybeLastDocument =>
+      maybeLastDocument.fold(countZero) { lastDocument =>
+        for {
+          documents <- persistenceDAO.getAllSince(lastDocument.upsertId, index)
+          count <- updateAndCount(documents, index)
+        } yield {
+          count
+        }
+      }
+    }
+  }
+
+  /**
+    * Apply a sequence of partial documents to an Elasticsearch index,
+    * returning the total number of documents applied.
+    *
+    * Updates run serially, and short-circuit if applying any one
+    * update fails.
+    */
+  private def updateAndCount[D <: ClioDocument: Indexable](
+    documents: Seq[D],
+    index: ElasticsearchIndex[D]
+  )(implicit ec: ExecutionContext): Future[Long] = {
+    documents.foldLeft(countZero) {
+      case (countFut, document) => {
```

Comment:
Where does the convention to abbreviate `future` as `fut` come from? I find that abbreviations in general make the code harder to read, and in this case it's only saving us three characters.